SpatialGDK/Source/SpatialGDK/Private/Utils/InterestFactory.cpp

```diff
@@ -239,6 +243,14 @@ Interest InterestFactory::CreateInterest(AActor* InActor, const FClassInfo& InIn
 		AddClientPlayerControllerActorInterest(ResultInterest, InActor, InInfo);
 	}
 
+#if WITH_UNREAL_DEVELOPER_TOOLS || (!UE_BUILD_SHIPPING && !UE_BUILD_TEST)
+	if (InActor->IsA(AGameplayDebuggerCategoryReplicator::StaticClass()))
```

Comment:
Rather do the isa/cast conversion twice, this perhaps could be written as:

```
if (AGameplayDebuggerCategoryReplicator* Replicator = Cast<AGameplayDebuggerCategoryReplicator>(InActor))
{
    AddServerGameplayDebuggerCategoryReplicatorActorInterest(ResultInterest, *Replicator);
}

```
This has the addition benefit of being a safe usage of a potentially null InActor pointer, which unfortunately the rest of this function isn't.

Reply:
changed